cmd/noms/noms.go

```diff
@@ -86,7 +105,128 @@ func main() {
 			return
 		}
 	}
+}
+
+var (
```

Comment:
The globals here skeeve me out. How about having caller just track these pointers itself?